src/test/scala/io/iohk/ethereum/vm/SSTOREOpCodeGasPostConstantinopleSpec.scala

```diff
@@ -10,47 +10,76 @@ import akka.util.ByteString.{empty => bEmpty}
-    ("60026000556000600055", 1, 5212, 15000),
-    ("60026000556003600055", 1, 5212, 0),
-    ("60026000556001600055", 1, 5212, 4800),
-    ("60026000556002600055", 1, 5212, 0),
-    ("60016000556000600055", 1, 5212, 15000),
-    ("60016000556002600055", 1, 5212, 0),
-    ("60016000556001600055", 1, 412, 0),
-    ("600160005560006000556001600055", 0, 40218, 19800),
-    ("600060005560016000556000600055", 1, 10218, 19800)
-  )
-
-
+  // Spec https://eips.ethereum.org/EIPS/eip-1283
   "Net gas metering for SSTORE after Constantinople hard fork (EIP-1283)" in {
-    forAll(table) {
+    val eip1283table =  Table[String, BigInt, BigInt, BigInt](
+      ("code", "original", "gasUsed", "refund"),
+      ("60006000556000600055", 0, 412, 0),
+      ("60006000556001600055", 0, 20212, 0),
+      ("60016000556000600055", 0, 20212, 19800),
+      ("60016000556002600055", 0, 20212, 0),
+      ("60016000556001600055", 0, 20212, 0),
+      ("60006000556000600055", 1, 5212, 15000),
+      ("60006000556001600055", 1, 5212, 4800),
+      ("60006000556002600055", 1, 5212, 0),
+      ("60026000556000600055", 1, 5212, 15000),
+      ("60026000556003600055", 1, 5212, 0),
+      ("60026000556001600055", 1, 5212, 4800),
+      ("60026000556002600055", 1, 5212, 0),
+      ("60016000556000600055", 1, 5212, 15000),
+      ("60016000556002600055", 1, 5212, 0),
+      ("60016000556001600055", 1, 412, 0),
+      ("600160005560006000556001600055", 0, 40218, 19800),
+      ("600060005560016000556000600055", 1, 10218, 19800)
+    )
+
+    forAll(eip1283table) {
       (code, original, gasUsed, refund) => {
-        val result = vm.exec(prepareProgramState(ByteString(Hex.decode(code)), original))
+        val result = vm.exec(prepareProgramState(ByteString(Hex.decode(code)), original, EipToCheck.EIP1283))
 
         result.gasUsed shouldEqual gasUsed
         result.gasRefund shouldEqual refund
       }
     }
   }
 
+  // Spec https://eips.ethereum.org/EIPS/eip-2200
+  "Net gas metering for SSTORE after Phoenix hard fork (EIP-2200)" in {
+    val eip2200table =  Table[String, BigInt, BigInt, BigInt](
+      ("code", "original", "gasUsed", "refund"),
+      ("60006000556000600055", 0, 1612, 0),
+      ("60006000556001600055", 0, 20812, 0),
+      ("60016000556000600055", 0, 20812, 19200),
+      ("60016000556002600055", 0, 20812, 0),
+      ("60016000556001600055", 0, 20812, 0),
+      ("60006000556000600055", 1, 5812, 15000),
+      ("60006000556001600055", 1, 5812, 4200),
+      ("60006000556002600055", 1, 5812, 0),
+      ("60026000556000600055", 1, 5812, 15000),
+      ("60026000556003600055", 1, 5812, 0),
+      ("60026000556001600055", 1, 5812, 4200),
+      ("60026000556002600055", 1, 5812, 0),
+      ("60016000556000600055", 1, 5812, 15000),
+      ("60016000556002600055", 1, 5812, 0),
+      ("60016000556001600055", 1, 1612, 0),
+      ("600160005560006000556001600055", 0, 40818, 19200),
+      ("600060005560016000556000600055", 1, 10818, 19200)
```

Comment:
I found out that geth has two additional test cases : https://github.com/ethereum/go-ethereum/blob/b9df7ecdc3d3685180ceb29665bab59e9f614da5/core/vm/gas_table_test.go#L76. 
```
	{1, 2306, "0x6001600055", 2306, 0, ErrOutOfGas},                            // 1 -> 1 (2300 sentry + 2xPUSH)
	{1, 2307, "0x6001600055", 806, 0, nil},                                     // 1 -> 1 (2301 sentry + 2xPUSH)
```
 Most probably testing some edge cases. Imo we should try to add them, as all this gass couting makes me nervous,

Reply:
Added